src/librustc_mir/const_eval.rs

```diff
@@ -694,8 +694,8 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
         if tcx.is_static(def_id).is_some() {
             let err = err.report_as_error(ecx.tcx, "could not evaluate static initializer");
             // check that a static never produces `TooGeneric`
-            if tcx.sess.err_count() == 0 {
-                span_bug!(ecx.tcx.span, "static eval failure didn't emit an error: {:#?}", err);
+            if let ErrorHandled::TooGeneric = err {
+                span_bug!(ecx.tcx.span, "static eval failure did not emit an error");
```

Comment:
I think we should rather change it to `ecx.tcx.sess.delay_span_bug(ecx.tcx.span, "...")`. We really don't want this to accidentally get accepted. If there's a layout size overflow, but no error reported, that would be bad, too.

Reply:
Ah, good point!